### PR TITLE
feat: add telemetry to status calls

### PR DESCRIPTION
**Pull request by pdgonzalez872**

### CODE REVIEW

File: lib/avalanche.ex [hunk 1]
```diff
@@ -194,14 +194,23 @@ defmodule Avalanche do
...
          {:ok, valid_request_opts} <- validate_options(request_opts, @request_options_schema),
          {:ok, valid_status_opts} <- validate_options(status_options, @status_options_schema),
          async <- Keyword.fetch!(valid_status_opts, :async),
-         partition <- Keyword.fetch!(valid_status_opts, :partition) do
-      statement_handle
-      |> Avalanche.StatusRequest.build(valid_request_opts)
-      |> Avalanche.StatusRequest.run(async, partition)
+         partition <- Keyword.fetch!(valid_status_opts, :partition),
+         metadata <- %{async: async, partition: partition} do
+      try do
```
Comment (pdgonzalez872) [inline lib/avalanche.ex:205]: due to the existing `with` I ended up adding the try/catch around the request piece. I didn't do it like this for statement and wonder if we should to make it consistent. Thoughts @forest ?

Comment (forest) [reply lib/avalanche.ex:205]: Why not add the `try` around the `with` like for `run`?

File: lib/avalanche.ex [hunk 2]
```diff
@@ -194,14 +194,23 @@ defmodule Avalanche do
...
     with request_opts <- Keyword.merge(default_options(), request_options),
          {:ok, valid_request_opts} <- validate_options(request_opts, @request_options_schema),
          {:ok, valid_status_opts} <- validate_options(status_options, @status_options_schema),
          async <- Keyword.fetch!(valid_status_opts, :async),
-         partition <- Keyword.fetch!(valid_status_opts, :partition) do
-      statement_handle
-      |> Avalanche.StatusRequest.build(valid_request_opts)
-      |> Avalanche.StatusRequest.run(async, partition)
+         partition <- Keyword.fetch!(valid_status_opts, :partition),
+         metadata <- %{async: async, partition: partition} do
```
Comment (forest) [inline lib/avalanche.ex:204]: I think you want `statement_handle` in the metadata too.

File: lib/avalanche.ex (hunk 1 above)
Comment (pdgonzalez872) [reply lib/avalanche.ex:205]: mostly because I'd need to take things out of the with. Nothing more. super quick change.

Comment (pdgonzalez872) [reply lib/avalanche.ex]: ✅ f4d7bc0ad3546240456c6c6f014d1570b6a64569

File: lib/avalanche.ex (hunk 2 above)
Comment (pdgonzalez872) [reply lib/avalanche.ex]: ✅ f4d7bc0ad3546240456c6c6f014d1570b6a64569

File: lib/avalanche.ex [hunk 3]
```diff
@@ -194,14 +194,27 @@ defmodule Avalanche do
...
-         partition <- Keyword.fetch!(valid_status_opts, :partition) do
-      statement_handle
-      |> Avalanche.StatusRequest.build(valid_request_opts)
-      |> Avalanche.StatusRequest.run(async, partition)
+    start_time = System.monotonic_time()
+
+    metadata =
+      %{statement_handle: statement_handle}
+      |> Map.put(:async, Keyword.get(status_options, :async))
+      |> Map.put(:partition, Keyword.get(status_options, :partition))
```
Comment (forest) [inline lib/avalanche.ex:202]: Sorry to be a pain, but maybe put this `metadata` stuff in the catch since it isn't needed anywhere else.

Comment (pdgonzalez872) [reply lib/avalanche.ex]: haha, never a pain! Done!

**Review by forest — APPROVED** [review]

**Pull request merged**